server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -493,6 +513,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersions.V_8_13_0)) {
             out.writeOptionalBoolean(this.requireDataStream);
         }
+        if (out.getTransportVersion().onOrAfter(TransportVersions.FAILURE_STORE_LAZY_CREATION)) {
+            out.writeOptionalBoolean(this.initializeFailureStore);
```

Comment:
This boolean should not be optional. You are using `in.readBoolean()` to read it. Same goes for the one above. 

I was really surprised this works to begin with. So, I looked up boolean serialisation and it's using `0` for `false`, `1` for `true` and `2` for `null`. This is why this code still works, while if it was a different optional object it would not. This code works because both `requireDataStream` and `initializeFailureStore` are primitives and `null` is not an option. But the code is still wrong and `readBoolean`, strictly expects `0` or `1`, if we ever pass `null` this will break.

Would you mind fixing both?